plugin/v1/plugin.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package plugin implements CloudKMS plugin for GKE as described in go/gke-secrets-encryption-design.
+package v1
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	"google.golang.org/api/cloudkms/v1"
+	"google.golang.org/grpc"
+
+	"github.com/GoogleCloudPlatform/k8s-cloudkms-plugin/plugin"
+)
+
+const (
+	netProtocol    = "unix"
+	apiVersion     = "v1beta1"
+	runtimeName    = "CloudKMS"
+	runtimeVersion = "0.0.1"
```

Comment:
This should probably come from the Go module now.

Reply:
The new KMS v2 API doesn't have this parameter in the plugin interface to pass.
This value was hardcoded in the v1 implementation.